storethehash.go

```diff
@@ -123,7 +135,7 @@ func (bs *HashedBlockstore) GetSize(c cid.Cid) (int, error) {
 
 // Put puts a given block to the underlying datastore
 func (bs *HashedBlockstore) Put(blk blocks.Block) error {
-	err := bs.store.Put(blk.Cid().Bytes(), blk.RawData())
+	err := bs.store.Put(blk.Cid().Hash(), blk.RawData())
```

Comment:
Be careful that the first few bits of the multihash are pure hash data, and not multihash algo identifier. Otherwise all multihashes, using the same type of hash will have the same prefix bits and will tank performance.  In the indexer core, this was a painful discovery (which you helped us understand), and the fix was [this](https://github.com/filecoin-project/go-indexer-core/blob/main/store/storethehash/storethehash.go#L499-L501).

Reply:
yea. this is actual block data, so CIDs should be pure, with no unexpected prefixes, and the multihash code decodes and uses only the digests.